### PR TITLE
fix: closed client should be able to reconnect

### PR DESCRIPTION
**Pull request by ronag**

### CODE REVIEW

**Comment by ronag** [comment]
added test

**Comment by ronag** [comment]
I've put a few commits into this PR to avoid the whole endless conflicts thing. The commits should be somewhat related. Hope you don't mind.

File: test/client-errors.js [hunk 1]
```diff
@@ -265,7 +265,7 @@ test('invalid URL throws', (t) => {
 })
 
 test('POST which fails should error response', (t) => {
-  t.plan(2)
+  t.plan(4)
```
Comment (mcollina) [inline test/client-errors.js:268]: why this change?

File: test/client.js [hunk 2]
```diff
@@ -586,3 +586,35 @@ test('close should call callback once finished', (t) => {
...
   })
 })
+
+test('close should still reconnect', (t) => {
+  t.plan(6)
+
+  const server = createServer((req, res) => {
+    setTimeout(function () {
+      res.end(req.url)
+    }, 10)
```
Comment (mcollina) [inline test/client.js:596]: please avoid the timeout, it would make the tests even more brittle

File: test/client-errors.js (hunk 1 above)
Comment (ronag) [reply test/client-errors.js:268]: It didn’t call all the calllbacks before

File: test/client.js (hunk 2 above)
Comment (ronag) [reply test/client.js]: removed, was not needed, came with copy paste of another test
